core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -55,14 +55,15 @@ object Partitioner {
    * We use two method parameters (rdd, others) to enforce callers passing at least 1 RDD.
    */
   def defaultPartitioner(rdd: RDD[_], others: RDD[_]*): Partitioner = {
-    val bySize = (Seq(rdd) ++ others).sortBy(_.partitions.length).reverse
-    for (r <- bySize if r.partitioner.isDefined && r.partitioner.get.numPartitions > 0) {
-      return r.partitioner.get
+    val rdds = Seq(rdd) ++ others
+    val filteredRdds = rdds.filter( _.partitioner.exists(_.numPartitions > 0 ))
```

Comment:
@srowen thank you very much , i am a new hand about saprk, but i'm interested in it very much,i have fixed my code style.thanks.